src/app/editor/contextualListener.js

```diff
@@ -180,6 +129,59 @@ class ContextualListener {
     }
     this._activeHighlights = []
   }
+
+  // GET GAS ESTIMATION
+  _getGasEstimation (contract) {
+    this.contract = this.results.data.contracts[this.results.source.target][contract.attributes.name]
+    this.estimationObj = this.contract.evm.gasEstimates
+    this.creationCost = this.estimationObj.creation.totalCost
+    this.codeDepositCost = this.estimationObj.creation.codeDepositCost
+  }
+
+  gasEstimation (node) {
+    this._getContract(node)
+    var executionCost
+    var codeDepositCost
+    if (node.name === 'FunctionDefinition') {
+      if (!node.attributes.isConstructor) {
+        var fnName = node.attributes.name
+        var fn = fnName + this._getInputParams(fnName)
+        if (node.attributes.visibility === 'public') {
+          executionCost = this.estimationObj.external[fn]
+        } else if (node.attributes.visibility === 'internal') {
+          executionCost = this.estimationObj.internal[fn]
+        }
+      } else {
+        executionCost = this.creationCost
+        codeDepositCost = this.codeDepositCost
+      }
+    } else {
+      executionCost = '-'
+    }
+    return {executionCost, codeDepositCost}
+  }
+
+  _getContract (node) {
+    for (var i in this.nodes) {
+      if (this.nodes[i].id === node.attributes.scope) {
+        var contract = this._getGasEstimation(this.nodes[i])
+        break
+      }
+    }
+    return contract
+  }
+
+  _getInputParams (fnName) {
+    var abi = this.contract.abi
```

Comment:
that 'll not help to use the abi here, you have to loop over InputParameter of the FunctionDefinition AST node.